src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,33 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
 
 public class AsyncRequestProcessor {
+    private static final String USER_DETAILS = "Details for %s";
+
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(() -> getUserData(userId), executor);
+    }
+
+    private UserData getUserData(String userId) {
+        try {
+            long timeout = ThreadLocalRandom.current().nextLong(1000);
+            TimeUnit.MILLISECONDS.sleep(timeout);
+        } catch (InterruptedException ex) {
+            Thread.currentThread().interrupt();
+        }
+        return cache.computeIfAbsent(userId, s -> new UserData(s, USER_DETAILS.formatted(s)));
```

Comment:
what is the point of our cache if you are sleeping the thread without considering that we have data in it or not? we need sleep here to simulate db access of data that we have not fetched yet, and should use cache without the sleep if we already have it